uqcsbot/util/attachments_util.py

```diff
@@ -0,0 +1,254 @@
+        self.button_style = ButtonStyle.STYLE_PRIMARY.value
+
+    def set_button_style(self,bs:ButtonStyle):
+        self.button_style = bs.value
+
+class AttachmentField:
+    field_title:str
+    field_value:str
+
+    # Optional
+    # This param specifies if the field is skinny
+    #  enough to be shown beside others
+    field_short:bool = False
+
+    def __init__(self, title:str, value:str, short:Optional[bool]):
+        self.field_title = title
+        self.field_value = value
+        if not short == None:
+            self.field_short = short
+
+
+class AttachmentColor(Enum):
+    COLOUR_GOOD = "good"
+    COLOUR_WARNING = "warning"
+    COLOUR_DANGER = "danger"
+
+# https://api.slack.com/docs/message-attachments
+class Attachment:
+    # Non Optional Params (can be "")
+    attachment_fallback:str
+    attachment_text:str
+
+    # Optional Params
+    attachment_color:str = None
+    attachment_pretext:str = None
+    attachment_title:str = None
+    # Supplying the link will automatically hyperlink the title
+    attachment_title_link:str = None
+    # Should have at most 2 or 3 (as per the docs)
+    attachment_fields:List[AttachmentField] = []
+    attachment_footer:str = None
+    attachment_actions:List[AttachmentAction] = [] # At most 5
+
+    # Unimplemented Params
+    # Author paramaters
+    # Image URL
+    # Thumb URL
+    # Footer Icon
+    # ts (timestamp)
+
+    def __init__(self, fallback:str, text:str):
+        self.attachment_fallback = fallback
+        self.attachment_text = text
+
+    def set_color(self, ac:AttachmentColor):
+        self.attachment_color = ac.value
+
+    def validate(self) -> bool:
+        if not self.attachment_title_link == None:
+            if self.attachment_title == None:
+                # Should only define the link when the title is set
+                return False
+            if not validateURL(self.attachment_title_link):
+                # Attachment Link should be valid if not None
+                return None
+        if not self.attachment_fields == []:
+            # There is the strongly suggested limit of 3
+            # This limit could be removed if needs be,
+            # or some strict_validate could be added
+            # to ignore this item
+            if len(self.attachment_fields) > 3:
+                return False
+        if not self.attachment_actions == []:
+            actionsCount:int = len(self.attachment_actions)
+            # Should not add empty actions list
+            if actionsCount == 0:
+                return False
+            # Should not have more than 5 actions
+            if actionsCount > 5:
+                return False
+
+            # Validates each of the actions to make sure they are valid
+            act:AttachmentAction
+            for act in self.attachment_actions:
+                if not act.validate:
+                    return False
+
+        return True
+    
+
+class Attachments_Util:
```

Comment:
Convention in Python is no underscores, upper camel case for class names. Also please explicitly specify subclassing from `object`.